src/Moryx.Products.Management/Implementation/RecipeStorage.cs

```diff
@@ -133,18 +133,16 @@ private static List<IWorkplanStep> LoadSteps(WorkplanEntity workplan, IDictionar
                 }
 
                 // Restore parameters from JSON
-                var taskStep = step as ITaskStep<IParameters>;
-                if (taskStep != null)
-                {
+                if (step is ITaskStep<IParameters> taskStep)
                     JsonConvert.PopulateObject(stepEntity.Parameters, taskStep.Parameters);
-                }
 
                 // Restore Subworkplan if necessary
-                var subworkplanStep = step as ISubworkplanStep;
-                if (subworkplanStep != null)
-                {
+                if (step is ISubworkplanStep subworkplanStep)
                     subworkplanStep.Workplan = LoadWorkplan(stepEntity.SubWorkplan);
-                }
+
+                // Restore step name
+                if (step is INamedTaskStep namedTaskStep)
```

Comment:
Maybe extend the condition if the entity value is not `NullOrEmpty`. In that case we should keep the constructor initialized Name

Reply:
done